.github/workflows/devcontainer-feature-release.yml

```diff
@@ -0,0 +1,27 @@
+name: Build dev container feature image
+
+env:
+  IMAGE_NAME: ${{ github.repository }}
+  REGISTRY: ghcr.io
+
+on:
+  push:
```

Comment:
I think the release workflow should depend on successful run of the test workflow. Do we want the release process to be manual or automatic based on pushes to the relevant path? If automatic, what happens if there's a change to the relevant paths but the version number in `devcontainer-feature.json` isn't updated? (Does the existing version get overwritten?)

Given these I suggest: 

* Move the `deploy` job into `devcontainer-feature-test.yml` and only run the job in the event of a `workflow_dispatch` trigger (probably with a checkbox parameter like `Release container` defaulted to `off`)
* Make the `deploy` job depend on all `test*` jobs running successfully
* If the version is overwritten on release we should probably have some validation that we aren't overwriting the existing version
* On release we should increment the version number of the feature in `devcontainer-feature.json` and open a PR to do that updating

Reply:
Daniel will: 

* add version incrementing script, wire up to release

Aaron will: 

* require tests to pass before deploying 
* verify that version won't be overwritten
* remove `push` trigger to prevent unintended releases

Reply:
Use environment `deploy-devcontainer-feature`

Reply:
After further discussion we don't need to do version automation on this package. We don't expect there to be many changes that require automated releases.

Reply:
Only open discussion item at this point: Should publishing happen through GitHub Actions (I think yes but need to double check that)